### PR TITLE
Vyu/tdi 45797 s3put enable object lock feature

### PR DESCRIPTION
**Pull request by vyu-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-45797

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tS3Put/tS3Put_begin.javajet [hunk 1]
```diff
@@ -130,7 +136,25 @@ accessCtrlList.grantPermission(grantee_<%=cid%>_<%=grantee_id%>, com.amazonaws.s
 			}
 		}
   		%>
+		boolean enableObjectLock_<%=cid %> = <%=enableObjectLock %>;
+		String retentionMode_<%=cid %> = "<%=retentionMode %>";
```
Comment (pyzhou-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tS3Put/tS3Put_begin.javajet:140]: Why convert this variable from javajet into generated code? Does it have a special reason?

Comment (vyu-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tS3Put/tS3Put_begin.javajet:140]: Just want users can read the code clearly.

Comment (pyzhou-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tS3Put/tS3Put_begin.javajet:140]: The customer should rarely read the code. The logic of generate code goes into javajet, in terms of **HIDING** this logic from the customer. less is more.

**Review by pyzhou-talend — APPROVED** [review]
LGTM

**Review by pyzhou-talend — APPROVED** [review]
LGTM

**Review by jzhao-talend — CHANGES_REQUESTED** [review]
https://github.com/Talend/tdi-studio-se/blob/vyu/TDI-45797_s3put_enable_object_lock_feature/main/plugins/org.talend.designer.components.localprovider/components/tS3Copy/tS3Copy_begin.javajet#L92
Your changes would break tS3Copy

**Pull request merged**